app/code/Magento/CatalogGraphQl/Model/Resolver/Products/SearchCriteria/CollectionProcessor/FilterProcessor/CategoryFilter.php

```diff
@@ -61,7 +61,6 @@ public function apply(Filter $filter, AbstractDb $collection)
             $category = $this->categoryFactory->create();
             $this->categoryResourceModel->load($category, $categoryId);
             $categoryProducts[$categoryId] = $category->getProductCollection()->getAllIds();
-            $collection->addCategoryFilter($category);
```

Comment:
Hi @sudheers-kensium. Thank you for your contribution. I'm not sure that by removing this filter we won't introduce an additional issue. This filter has been recently added for solving the sorting issue. Details are below

https://github.com/magento/magento2/commit/dfcf474d83c05620e6fa3010ae25632cb820174e#diff-8a27c9476b188f4e8830e9872984bdfbR64

@cpartica could you provide more information on the consequences, please? Unfortunately, I have a lack of details about MC-32278. 

Thank you!

Reply:
@rogyar I have changed the code. Please check once.